mesonbuild/backend/backends.py

```diff
@@ -437,22 +438,13 @@ def init_language_file(suffix: str, unity_file_number: int) -> T.TextIO:
 
         # For each language, generate unity source files and return the list
         for comp, srcs in compsrcs.items():
-            files_in_current = unity_size + 1
-            unity_file_number = 0
-            # TODO: this could be simplified with an algorithm that pre-sorts
-            # the sources into the size of chunks we want
-            ofile = None
-            for src in srcs:
-                if files_in_current >= unity_size:
-                    if ofile:
-                        ofile.close()
-                    ofile = init_language_file(comp.get_default_suffix(), unity_file_number)
-                    unity_file_number += 1
-                    files_in_current = 0
-                ofile.write(f'#include<{src}>\n')
-                files_in_current += 1
-            if ofile:
-                ofile.close()
+            chunks = [srcs] if unity_size == -1 else grouper(srcs, unity_size)
+            for unity_file_number, chunk in enumerate(chunks):
+                with init_language_file(comp.get_default_suffix(), unity_file_number) as f:
+                    for src in chunk:
```

Comment:
We tend to not use `filter` since Python considers it semi-deprecated. I also find that particular construct so arcane I wouldn't use it, so by the time you write either `for src in filter(lambda x: x is not None, chunk)` or `for src in (s for s in chunk if s is not None)` it doesn't seem like a win.